app/controllers/attendances_controller.rb

```diff
@@ -0,0 +1,58 @@
+class AttendancesController < ApplicationController
+  before_action :set_user, :set_group, :authenticate_user!
+  before_action :set_attendance, only: [:show]
+
+  def index
+    if params[:date].present?
```

Comment:
we shouldn't be able to reach this page withoud a date, probably add a validation that disables the link/button to this page until you select a date, this way we get rid off all these conditionals